searchlib/src/vespa/searchlib/features/dotproductfeature.cpp

```diff
@@ -830,38 +789,87 @@ createQueryVector(const IQueryEnvironment & env, const IAttributeVector * attrib
         Property prop = env.getProperties().lookup(baseName, queryVector);
         if (prop.found() && !prop.get().empty()) {
             if (attribute->isStringType() && attribute->hasEnum()) {
-                dotproduct::wset::EnumVector vector(attribute);
-                WeightedSetParser::parse(prop.get(), vector);
+                auto vector = std::make_unique<dotproduct::wset::EnumVector>(attribute);
+                WeightedSetParser::parse(prop.get(), *vector);
+                arguments = std::move(vector);
             } else if (attribute->isIntegerType()) {
                 if (attribute->hasEnum()) {
-                    dotproduct::wset::EnumVector vector(attribute);
-                    WeightedSetParser::parse(prop.get(), vector);
+                    auto vector = std::make_unique<dotproduct::wset::EnumVector>(attribute);
+                    WeightedSetParser::parse(prop.get(), *vector);
+                    arguments = std::move(vector);
                 } else {
-                    dotproduct::wset::IntegerVector vector;
-                    WeightedSetParser::parse(prop.get(), vector);
+                    auto vector = std::make_unique<dotproduct::wset::IntegerVector>();
+                    WeightedSetParser::parse(prop.get(), *vector);
+                    arguments = std::move(vector);
```

Comment:
createFromObject seems to not be able to handle this

Reply:
It does now.